build.gradle.kts

```diff
@@ -0,0 +1,210 @@
+    compile("io.github.microutils", "kotlin-logging", "1.5.3")
+
+    // JavaFX
+    compile("de.jensd", "fontawesomefx", "8.9")
+    compile("org.reactfx", "reactfx", "2.0-M5")
+    compile("eu.hansolo", "tilesfx", "1.5.2")
+    compile("eu.hansolo", "Medusa", "7.9")
+
+    // Player
+    compile("uk.co.caprica", "vlcj", "3.10.1")
+    compile("de.u-mass", "lastfm-java", "0.1.2")
+
+    // annotation processor
+    compile("org.atteo.classindex", "classindex", "3.4")
+    apt("org.atteo.classindex", "classindex", "3.4")
+
+    // misc
+    compile("com.adobe.xmp", "xmpcore", "5.1.2")
+    compile("com.drewnoakes", "metadata-extractor", "2.9.1")
+    compile("net.objecthunter", "exp4j", "0.4.8")
+    compile("org.atteo", "evo-inflector", "1.2.2")
+    compile("com.thoughtworks.xstream", "xstream", "1.4.10")
+
+    //	compile("com.twelvemonkeys.imageio", "imageio-core", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-bmp", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-jpeg", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-iff", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-icns", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pcx", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pict", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-clippath", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-hdr", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pdf", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pnm", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-psd", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tga", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-sgi", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-thumbsdb", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tiff", "3.3.2")
+
+    compile(files(file("extra/lib").listFiles()))
+
+}
+
+// source directories
+java.sourceSets {
+    getByName("main") {
+        java.srcDir("src")
+        resources.srcDir("src")
+    }
+}
+
+tasks {
+    val main = "_Main"
+
+    withType<JavaCompile> {
+        options.isWarnings = false
+        options.isDeprecation = false
+        // javac args
+        options.compilerArgs = listOf("-Xlint:unchecked", "-nowarn",
+                "--add-exports", "javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.javafx.scene.traversal=ALL-UNNAMED",
+                "--add-exports", "javafx.web/com.sun.webkit=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.glass.ui=ALL-UNNAMED")
+    }
+
+    "compileJava" {
+        doFirst {
+            println("Java version: ${JavaVersion.current()}")
+        }
+    }
+
+    "compileKotlin" {
+        doFirst {
+            println("Kotlin version: $kotlinVersion")
+        }
+    }
+
+    "build" {
```

Comment:
I think the resulting jar should be generated to working directory and launched from there. Even when developing. This would simplify things.

Reply:
Doesn't the run task suffice for now? But yeah I could easily do that

Reply:
The idea is for the deployment step to not require any work (like copying the app jar). The reason is simple: I want to make the developer be able to bypass the gradle as much as possible if he wants to - the less he is required to know the better.

Reply:
Yes there will be a release task soon, but I can't do everything at once ;)
first we gotta clear up the whole question of vlc